h5py/_hl/files.py

```diff
@@ -443,6 +467,9 @@ def __init__(self, name, mode='r', driver=None,
             if fs_strategy and mode not in ('w', 'w-', 'x'):
                 raise ValueError("Unable to set file space strategy of an existing file")
 
+            if page_buf_size and mode not in ('r', 'r+'):
```

Comment:
Does it really make sense to only set this for these modes? It sounds from the HDF5 docs like the page buffer also applies to writing, and you can read data in any mode.

Reply:
Correct, I was given misleading info. Removed this check.